src/theme/ThemeProvider.js

```diff
@@ -0,0 +1,84 @@
+import React, {
+  useState,
+  useLayoutEffect,
+  useMemo,
+  createContext
+} from "react";
+import PropTypes from "prop-types";
+
+export const ThemeContext = createContext();
+
+export const ThemeProvider = ({
+  themes,
+  defaultThemeName,
+  fontConfig,
+  children
+}) => {
+  const [themeName, setThemeName] = useState(defaultThemeName);
+  const currentTheme = useMemo(() => themes[themeName], [themes, themeName]);
+  useLayoutEffect(() => {
+    if (currentTheme) {
+      Object.keys(currentTheme).forEach((key) => {
+        document.documentElement.style.setProperty(
+          `--${key}`,
+          currentTheme[key]
+        );
+      });
+      if (fontConfig) {
+        applyFonyAssets(fontConfig, currentTheme);
+      }
+    }
+  }, [currentTheme, fontConfig]);
+
+  return (
+    <ThemeContext.Provider
+      value={{
+        useThemeName: [themeName, setThemeName],
+        currentTheme
+      }}>
+      {children}
+    </ThemeContext.Provider>
+  );
+};
+
+ThemeProvider.propTypes = {
+  themes: PropTypes.object.isRequired,
+  defaultThemeName: PropTypes.string.isRequired,
```

Comment:
Probably minor but you could expand the validation of this prop by doing something like:

```
ThemeProvider.propTypes = {
    theme: PropTypes.object.isRequired,
    defaultThemeName: function(props, propName, componentName) {
         // check that the defaultThemeName is equal to one of the keys of themes
         // and throws an error if it is not
     }
    }
}
```